internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -380,6 +389,8 @@ func getComponents(spec *aiv1beta1.AgentServiceConfigSpec, isOpenshift bool) []c
 				component{"AgentIPXERoute", aiv1beta1.ReasonAgentRouteFailure, newAgentIPXERoute},
 			)
 		}
+	} else {
+		components = append(components, certManagerComponents()...)
```

Comment:
If we extract this we can return early and avoid the nesting and else condition:

```golang
components = append(components, getExtraComponents(spec))
```

Reply:
Not sure I see the benefit. Then `getExtraComponents` will still have the `if isOpenshift ... else ...`, right? Plus the logic isn't particularly complicated and the new function wouldn't be reused so it's not something I would think of breaking out for these reasons.